core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -217,22 +215,20 @@ protected void add(DataFile file) {
         // open all of the manifest files in parallel, use index to avoid reordering
         Tasks.range(filtered.length)
             .stopOnFailure().throwFailureWhenFinished()
-            .executeWith(getWorkerPool())
+            .executeWith(ThreadPools.getWorkerPool())
             .run(index -> {
-              ManifestFile manifest = filterManifest(
-                  deleteExpression, metricsEvaluator,
-                  manifests.get(index));
+              ManifestFile manifest = filterManifest(metricsEvaluator, manifests.get(index));
               filtered[index] = manifest;
             }, IOException.class);
 
         for (ManifestFile manifest : filtered) {
-          PartitionSpec spec = ops.current().spec(manifest.partitionSpecId());
+          PartitionSpec manifestSpec = ops.current().spec(manifest.partitionSpecId());
           Iterable<DataFile> manifestDeletes = filteredManifestToDeletedFiles.get(manifest);
           if (manifestDeletes != null) {
-            for (DataFile file : manifestDeletes) {
-              summaryBuilder.deletedFile(spec, file);
+            manifestDeletes.forEach(file -> {
```

Comment:
This is another change that looks like it's "fixing" cyclomatic complexity. Can we refactor this in a way that doesn't use lambdas as well?